### PR TITLE
Update content build image in CircleCI

### PR DESCRIPTION
**Pull request by ilaner**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-5893

## Description
* Instead of using our Content-Build image, use official CircleCI image (cimg)
* Install poetry, python virtualenv, node with official CircleCI tools
## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -55,6 +70,33 @@ references:
       PULL_REQUEST_NUMBER: << pipeline.parameters.pr_number >>
       NIGHTLY_PARAMETER: << pipeline.parameters.nightly >>
       GCS_MARKET_BUCKET: << pipeline.parameters.gcs_market_bucket >>
+  
+  install_poetry: &install_poetry
+    run:
+      name: Install Poetry
+      command: |
+        # in old images we need to remove existing poetry
```
Comment (esharf) [inline .circleci/config.yml:78]: why should we care about old images?

File: .circleci/config.yml [hunk 2]
```diff
@@ -84,7 +126,7 @@ references:
 
   container_config: &container_config
     docker:
-      - image: devdemisto/content-build:3.0.0.33772  # disable-secrets-detection
+      - image: cimg/python:3.10-node # disable-secrets-detection
```
Comment (esharf) [inline .circleci/config.yml:129]: we should use our images.
if you want to use another image, please open a PR in the dockerfiles repo.

**Review by esharf — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (ilaner) [reply .circleci/config.yml:129]: Why should we use our images?
The recommendation for CircleCI is to use the official CircleCI images.
The dependencies on top of this image could be downloaded with CircleCI official orb tools, with their cache.

Comment (ilaner) [reply .circleci/config.yml]: Edited the content-build image to be based on `cimg/python:3.10-node`

File: .circleci/config.yml (hunk 1 above)
Comment (ilaner) [reply .circleci/config.yml]: We don't. Removed this step

**Review by dorschw — APPROVED** [review]

**Pull request merged**